binding/react-native/ios/Porcupine.swift

```diff
@@ -9,137 +9,164 @@
-            &porcupine);
-
-        if status == PV_STATUS_SUCCESS {
-            let handle:String = String(describing:porcupine)
-            porcupinePool[handle] = porcupine;
+        do {
+            let porcupine = try Porcupine(
+                accessKey: accessKey,
+                keywordPaths: keywordPaths,
+                modelPath: modelPath.isEmpty ? nil : try getResourcePath(modelPath),
+                sensitivities: sensitivities.isEmpty ? nil : sensitivities)
             
-            let porcupineParameters:Dictionary<String, Any> = [
-                "handle": handle,
-                "frameLength": UInt32(pv_porcupine_frame_length()),
-                "sampleRate": UInt32(pv_sample_rate()),
-                "version": String(cString: pv_porcupine_version())
-            ]
-            resolve(porcupineParameters)
-        }
-        else {
-            reject(pvStatusToExceptionCode(status), "Failed to initialize Porcupine", nil)
+            let handle: String = String(describing: porcupine)
+            porcupinePool[handle] = porcupine
+            
+            var param: [String: Any] = [:]
+            param["handle"] = handle
+            param["frameLength"] = Porcupine.frameLength
+            param["sampleRate"] = Porcupine.sampleRate
+            param["version"] = Porcupine.version
+            
+            resolve(param)
+        } catch let error as PorcupineError {
+            let (code, message) = errorToCodeAndMessage(error)
+            reject(code, message, nil)
+        } catch {
+            let (code, message) = errorToCodeAndMessage(PorcupineError.PorcupineError(error.localizedDescription))
+            reject(code, message, nil)
         }
     }
     
     @objc(delete:)
     func delete(handle:String) -> Void {        
-        if var porcupine = porcupinePool.removeValue(forKey: handle){
-            pv_porcupine_delete(porcupine)
-            porcupine = nil
+        if let porcupine = porcupinePool.removeValue(forKey: handle) {
+            porcupine.delete()
         }
     }
     
     @objc(process:pcm:resolver:rejecter:)
     func process(handle:String, pcm:[Int16], 
         resolver resolve:RCTPromiseResolveBlock, rejecter reject:RCTPromiseRejectBlock) -> Void {
-                
-        if let porcupine = porcupinePool[handle]{
-            var keywordIndex: Int32 = -1
-            pv_porcupine_process(porcupine, pcm, &keywordIndex)        
-            
-            resolve(keywordIndex)
-        }
-        else{
-            reject(pvStatusToExceptionCode(PV_STATUS_INVALID_STATE), "Invalid Porcupine handle provided to native module.", nil)
+        do {
+            if let porcupine = porcupinePool[handle] {
+                let keywordIndex = try porcupine.process(pcm: pcm)
+                resolve(keywordIndex)
+            } else {
+                let (code, message) = errorToCodeAndMessage(PorcupineError.PorcupineRuntimeError("Invalid handle provided to Porcupine 'process'"))
```

Comment:
I think we were doing InvalidState errors for when the object has been deleted and was being used again